concordium-node/src/bin/cli.rs

```diff
@@ -485,36 +485,70 @@ async fn maybe_do_out_of_band_catchup(
             info!("Completed out of band catch-up from {}.", import_blocks_from.display());
         }
     } else if let Some(download_url) = download_blocks_from.as_ref().cloned() {
-        let from = download_catchup_files(download_url.clone(), data_dir_path).await;
-        match from {
-            Ok(path) => {
-                info!("Starting out of band catch-up");
-                if let Err(e) = consensus.import_blocks(&path) {
-                    if import_stopped.load(atomic::Ordering::Acquire) {
-                        info!("Out of band catchup stopped.");
-                    } else {
-                        info!("Could not complete out of band catch-up due to: {:#}.", e);
-                    }
-                } else {
-                    info!("Completed out of band catch-up from {}.", download_url);
-                }
-                // attempt to properly clean up the downloaded file.
-                if let Err(e) = path.close() {
-                    error!("Could not delete the downloaded file: {}", e);
-                }
-            }
-            Err(e) => {
-                // report error, but continue startup
-                error!("Downloading catch-up files failed: {}.", e)
+        info!("Starting out of band catch-up");
+        if let Err(e) = import_missing_blocks(consensus, download_url, data_dir_path).await {
+            if import_stopped.load(atomic::Ordering::Acquire) {
+                info!("Out of band catchup stopped.");
+            } else {
+                info!("Could not complete out of band catch-up due to: {:#}.", e);
             }
+        } else {
+            info!("Completed out of band catch-up from {}.", download_url)
         }
     }
 }
 
-async fn download_catchup_files(
-    download_url: url::Url,
+#[derive(serde::Deserialize)]
+struct BlockChunkData {
+    filename:           String,
+    genesis_index:      u64,
+    first_block_height: u64,
+    last_block_height:  u64,
+}
+
+async fn import_missing_blocks(
+    consensus: &ConsensusContainer,
+    index_url: &url::Url,
     data_dir_path: &Path,
-) -> anyhow::Result<TempPath> {
+) -> anyhow::Result<()> {
+    let json_consensus_value: serde_json::Value =
+        serde_json::from_str(&consensus.get_consensus_status())?;
+    let best_block_height = json_consensus_value["bestBlockHeight"].as_u64().unwrap();
```

Comment:
We should add an ffi call to get the (current era) block height of the last finalized block.